subsys/net/lib/openthread/platform/radio.c

```diff
@@ -82,6 +85,8 @@ K_THREAD_STACK_DEFINE(ot_task_stack, OT_WORKER_STACK_SIZE);
 static struct k_work_q ot_work_q;
 static otError tx_result;
 
+K_MSGQ_DEFINE(rx_pkt_queue, sizeof(struct net_pkt *), OT_RX_MESSAGES, 4);
```

Comment:
Why message queue? We are passing net_pkt via fifo's in network stack so that could be used here too.

Reply:
Good point, that should simplify the code a bit.